.github/workflows/ci.yml

```diff
@@ -101,8 +119,20 @@ jobs:
         uses: actions/setup-go@v3
         with:
           go-version: ${{ env.go-version }}
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+          key: cache-go-${{ env.go-version }}-unittests-${{ hashFiles('**/go.sum') }}
```

Comment:
I think we can use the same cache name here as in the other steps, it's running on the same code as the other steps, no need to build a separate cache just for the unit-tests:

```yaml
key: cache-go-${{ env.go-version }}-${{ hashFiles('**/go.sum') }}
restore-keys: |
    cache-go-${{ env.go-version }}
```

Reply:
The problem is that multiple jobs using the same cache key that run **in parallel** race on storing the cache. It's OK for `quicktests` and `golangci-lint` as both just pull the dependency go modules. Whichever finishes first will store the cache. 

But `unittests` job also builds tests. As it takes the longest time to finish, the build cache would always be discarded, which is not what we want.  With a distinct key for `unittests`, its cache is always saved.